master/buildbot/buildslave/__init__.py

```diff
@@ -412,7 +412,7 @@ def _info_unavailable(why):
                 why.trap(pb.NoSuchMethod)
                 # maybe an old slave, doesn't implement remote_getSlaveInfo
                 log.msg("BuildSlave.info_unavailable")
-                log.err(why)
+                klog.err_json(why)
```

Comment:
Looks like we have many cases like this, calling `log.msg` and `log.err` later.

Perhaps we should add an optional parameter to `err_json`, eg:

`klog.err_json(why, header="BuildSlave.info_unavailable")`

And make that another field in the json dict, since it looks like that extra bit of information is always useful.

Reply:
err_json has already parameter `_why [string]`. Maybe we should use it instead new `header`

Reply:
Are you sure that's a string?

`why.trap(pb.NoSuchMethod)`

Doesn't look like it

Reply:
Look (from twisted/python/log.py)
```
def err(_stuff=None, _why=None, **kw):
...
    @param _why: The source of this failure.  This will be logged along with
        C{_stuff} and should describe the context in which the failure
        occurred.
    @type _why: C{str}
```

But in your example `why` is later a first argument for function
```
               why.trap(pb.NoSuchMethod)
                # maybe an old slave, doesn't implement remote_getSlaveInfo
                log.msg("BuildSlave.info_unavailable")
                klog.err_json(why)
```
So it is a _stuff [Failue/Exception].
In my opinion `why` in this example is a little confusing.

Reply:
Indeed, that was a terrible variable name at that point :P

Then I agree, using `why` sounds like what I wanted.

Reply:
Done. `_why` is added to final JSON as key `header`. 
If `_why` is None `header` will be `null`